lib/segment/src/id_tracker/mutable_id_tracker.rs

```diff
@@ -604,7 +604,12 @@ fn store_version_changes(
     })?;
 
     // Explicitly fsync file contents to ensure durability
-    let file = writer.into_inner().unwrap();
+    let file = writer.into_inner().map_err(|err| {
+        OperationError::service_error(format!(
+            "Failed to close ID tracker point versions write buffer: {}",
+            err.into_error()
+        ))
+    })?;
```

Comment:
@coszio Thanks for the review.

Though, this one is unnecessary. We've already completely flushed the writer right before it, so this cannot panic on unwrap. Sadly I need the unwrap here so that I can touch `sync_all()`.

Reply:
TBH, I'd prefer `?`. It's excessive, but it prevents accidental panics when someone would (intentionally or not) *remove* explicit flush, but miss your comment and unwrap.